### PR TITLE
community[minor]: Add embeddings integration for text2vec

### PR DESCRIPTION
**Pull request by hulitaitai**
Create a Class which allows to use the "text2vec" open source embedding model.

It should install the model by running 'pip install -U text2vec'.
Example to call the model through LangChain:

            from langchain_community.embeddings.text2vec import Text2vecEmbeddings

            embedding = Text2vecEmbeddings()
            bookend.embed_documents([
                "This is a CoSENT(Cosine Sentence) model.",
                "It maps sentences to a 768 dimensional dense vector space.",
            ])
            bookend.embed_query(
                "It can be used for text matching or semantic search."
            )

### CODE REVIEW

File: libs/community/langchain_community/embeddings/text2vec.py
Comment (eyurtsev) [inline libs/community/langchain_community/embeddings/text2vec.py:13]: 1) Could you provide a reference to the github repository for text2vec? So maintainers can inspect whether this is an OK dependency to add to the code base?
2) langchain_community uses optional imports exclusively -- imports should be in the local scope

for example: https://github.com/langchain-ai/langchain/blob/main/libs/community/langchain_community/chat_message_histories/redis.py#L28-L28

**Review by eyurtsev — CHANGES_REQUESTED** [review]
changes needed before can be merged

**Comment by eyurtsev** [comment]
Thanks for the contribution @hulitaitai , left a comments with next steps before we can merge your contribution

File: libs/community/langchain_community/embeddings/text2vec.py
Comment (hulitaitai) [reply libs/community/langchain_community/embeddings/text2vec.py:13]: The gitbub repository for text2vec is : 
https://github.com/shibing624/text2vec

The link of your example returns a 404 error.

Comment (hulitaitai) [reply libs/community/langchain_community/embeddings/text2vec.py:13]: What changes should I make to be merged? It is the second time that I write a little thing to pull the request. I fell first the lint check where I find nothing to change or improve in my code.

**Review by baskaryan — COMMENTED** [review]
can we add a docs page at `docs/docs/text_embedding/text2vec.ipynb`

File: libs/community/langchain_community/embeddings/text2vec.py
Comment (eyurtsev) [inline libs/community/langchain_community/embeddings/text2vec.py:36]:
```suggestion
        self, *, model: Any = None, model_name_or_path: Optional[str] = None, **kwargs: Any
```

File: libs/community/langchain_community/embeddings/text2vec.py
Comment (eyurtsev) [reply libs/community/langchain_community/embeddings/text2vec.py:12]: OK great! I didn't realize it was https://github.com/shibing624/text2vec. Yeah we can add it

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**
